skills_ml/storage/__init__.py

```diff
@@ -32,6 +32,13 @@ class S3Store(Store):
     def __init__(self, path):
         super().__init__(path=path)
 
+    @staticmethod
+    @contextmanager
+    def open(path, *args, **kwargs):
```

Comment:
I am confused at the use of the static method. It ignores the base path that the object has in it. This is fine if you are actually using it as a static method (e.g. `S3Store.open('...')`), but elsewhere in this pull request you are actually calling open as a method on instances of a storage object, which is confusing because it means the object does have a path which is being totally ignored in favor of this other path.

I would either:

1. Make `open` an instance method that works by concatenating the given filename to the base path, much like how the rest of the instance methods work.
2. Include an function that can be imported (like `open_sesame` in the matching tool) that lets us open files using full paths without the baggage of being attached to a storage object.

I think both #1 and #2 are truly useful, and in a standalone package version of this module i think both are worth including. But for our purposes here just picking one is probably enough.